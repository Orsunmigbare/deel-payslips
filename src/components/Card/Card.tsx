import { Payslip } from "types/api";
import styles from "./styles.module.css"
import { IonItem, IonImg } from "@ionic/react";

export const Card = (props: Payslip) => {
    return (
        <IonItem className={styles.cardCont} lines={"none"} detail={false} href={`/payslip/${props.id}`}>
            <div className={styles.card}>
                <div className={styles.info}>
                    <h3 className={styles.infoTitle}>Payslip for {new Date(props.fromDate).toDateString()} to {new Date(props.toDate).toDateString()}</h3>
                    <div className={styles.infoDesc}>Click to for details</div>
                </div>
                <div className={styles.preview}>
                    <figure>
                        <IonImg className={styles.previewImg}
                            src={props.file} alt="payslip"
                        ></IonImg>
                        <figcaption>Payslip for {new Date(props.fromDate).toDateString()} to {new Date(props.toDate).toDateString()} </figcaption>
                    </figure>
                </div>
            </div>
        </IonItem>
    )
}