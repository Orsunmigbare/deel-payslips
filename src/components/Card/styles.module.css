.cardCont {
    margin: 24px 0px 0px;
    border-radius: 8px;
    padding: 0;
    border: 1px solid #e8ebf2;
}

.cardCont .input-wrapper {
    display: block;
}

.card {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 24px;
    width: 100%;
}


.infoTitle {
    color: #444bc5;
    font-size: 14px;
    font-weight: 600;
    margin-bottom: 8px;
}

.infoDesc {
    color: #7c8286;
    font-size: 12px;
    line-height: 16px;
    font-weight: 400;
}

.preview {
    height: 50px;
    width: 50px;
    border-radius: 10px 10px 0 0;
    overflow: hidden;
    margin: 0 10px;
}

.previewImg {
    height: 80px;
    width: 80px;
}