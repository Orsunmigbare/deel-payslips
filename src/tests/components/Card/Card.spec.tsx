import "@testing-library/jest-dom"
import { render } from '@testing-library/react';
import { Card } from 'src/components/Card/Card';
import { SampleProps } from './data';


describe("Card", () => {
    it("renders correctly", () => {
        const screen = render(<Card {...SampleProps} />);
        expect(screen).toMatchSnapshot()
    })
    it("passes the right props", () => {
        const { getByText, getByAltText } = render(<Card {...SampleProps} />);
        const textContainingDate = `Payslip for ${new Date(SampleProps.fromDate).toDateString()} to ${new Date(SampleProps.toDate).toDateString()}`
        expect(getByText(textContainingDate)).toBeInTheDocument();
        expect(getByAltText("payslip")).toHaveAttribute("src", SampleProps.file);
        expect(document.querySelector("ion-item")).toHaveAttribute("href", `/payslip/${SampleProps.id}`)
    })
})