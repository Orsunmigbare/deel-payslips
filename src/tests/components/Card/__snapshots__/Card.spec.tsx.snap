// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ion-item
        class="cardCont"
        detail="false"
        href="/payslip/61f4a809a6c0b5d0e1840c11"
        lines="none"
      >
        <div
          class="card"
        >
          <div
            class="info"
          >
            <h3
              class="infoTitle"
            >
              Payslip for 
              Tue Feb 01 2022
               to 
              Tue Mar 01 2022
            </h3>
            <div
              class="infoDesc"
            >
              Click to for details
            </div>
          </div>
          <div
            class="preview"
          >
            <ion-img
              alt="payslip"
              class="previewImg"
              src="https://www.personalfinancelab.com/wp-content/uploads/h5p/content/13/images/image-5e502af138e62.PNG"
            />
          </div>
        </div>
      </ion-item>
    </div>
  </body>,
  "container": <div>
    <ion-item
      class="cardCont"
      detail="false"
      href="/payslip/61f4a809a6c0b5d0e1840c11"
      lines="none"
    >
      <div
        class="card"
      >
        <div
          class="info"
        >
          <h3
            class="infoTitle"
          >
            Payslip for 
            Tue Feb 01 2022
             to 
            Tue Mar 01 2022
          </h3>
          <div
            class="infoDesc"
          >
            Click to for details
          </div>
        </div>
        <div
          class="preview"
        >
          <ion-img
            alt="payslip"
            class="previewImg"
            src="https://www.personalfinancelab.com/wp-content/uploads/h5p/content/13/images/image-5e502af138e62.PNG"
          />
        </div>
      </div>
    </ion-item>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
